packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1955,51 +1844,55 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
-        const updatePayload: null | UpdatePayload = (finishedWork.updateQueue: any);
-        finishedWork.updateQueue = null;
-        if (updatePayload !== null) {
-          commitUpdate(
-            instance,
-            updatePayload,
-            type,
-            oldProps,
-            newProps,
-            finishedWork,
-          );
+      if (supportsMutation) {
+        const instance: Instance = finishedWork.stateNode;
+        if (instance != null) {
+          // Commit the work prepared earlier.
+          const newProps = finishedWork.memoizedProps;
+          // For hydration we reuse the update path but we treat the oldProps
+          // as the newProps. The updatePayload will contain the real change in
+          // this case.
+          const oldProps = current !== null ? current.memoizedProps : newProps;
+          const type = finishedWork.type;
+          // TODO: Type the updateQueue to be specific to host components.
+          const updatePayload: null | UpdatePayload = (finishedWork.updateQueue: any);
+          finishedWork.updateQueue = null;
+          if (updatePayload !== null) {
+            commitUpdate(
+              instance,
+              updatePayload,
+              type,
+              oldProps,
+              newProps,
+              finishedWork,
+            );
+          }
         }
       }
       return;
     }
     case HostText: {
-      if (finishedWork.stateNode === null) {
-        throw new Error(
-          'This should have a text node initialized. This error is likely ' +
-            'caused by a bug in React. Please file an issue.',
-        );
-      }
+      if (supportsMutation) {
+        if (finishedWork.stateNode === null) {
+          throw new Error(
+            'This should have a text node initialized. This error is likely ' +
+              'caused by a bug in React. Please file an issue.',
+          );
+        }
 
-      const textInstance: TextInstance = finishedWork.stateNode;
-      const newText: string = finishedWork.memoizedProps;
-      // For hydration we reuse the update path but we treat the oldProps
-      // as the newProps. The updatePayload will contain the real change in
-      // this case.
-      const oldText: string =
-        current !== null ? current.memoizedProps : newText;
-      commitTextUpdate(textInstance, oldText, newText);
+        const textInstance: TextInstance = finishedWork.stateNode;
+        const newText: string = finishedWork.memoizedProps;
+        // For hydration we reuse the update path but we treat the oldProps
+        // as the newProps. The updatePayload will contain the real change in
+        // this case.
+        const oldText: string =
+          current !== null ? current.memoizedProps : newText;
+        commitTextUpdate(textInstance, oldText, newText);
+      }
       return;
     }
     case HostRoot: {
-      if (supportsHydration) {
+      if (supportsMutation) {
```

Comment:
Shouldn't this be `supportsMutation && supportsHydration`? I guess it wouldn't hit `isDehydrated` in other renderers but might want to cut it out if third-party renderers run with GCC.

Reply:
Yeah sorry my brain keeps reading these as the same flag